CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# Contributing to the Lua Users Foundation
+
+This document serves as contribution guidelines and the points listed should be
+followed for contributions to the foundation.
+
+## Commit Guidelines
+
+As is somewhat standard practice, the following should be observed:
+
+- The usage of Markdown-formatted lists is allowed
+- Paragraphs should be broken with an empty line between
+- Further lines after the header should be no more than 72 characters
+- Further lines, if required, should contain detailed information on:
+ 1. Implications of the changes in the commit
+ 2. Reasons the changes are required
+
+Commits relating to one file should follow these guidelines:
+
+- The header line should be no more than 60 characters
+- The header line should contain the filename and a short description of what
+applying the patch will do, in the format of "If applied, this commit will..."
+- The second line should be blank
+
+Example commit message:
+
+```markdown
+CONTRIBUTING.md: Establish guidelines for new commits
+
+Implications of Changes:
+
+- Could possibly make it harder for outsiders to contribute
+- Could create bloatware in the repository data
+
+Reasons for Changes:
+
+- Creates a structure which can be used to avoid excessive comments
+  during the review process
+```
+
+Commits relating to multiple files should follow these guidelines:
```

Comment:
Kind of disagree we should treat commits using multiple files or single files differently. Looks like it could generate some inconsistency. Also I don't think there's any reason to have different max header length for these scenarios.

Reply:
I put a limit on max header length because GitHub itself even recommends that length, and will trim messages that it considers "too long".

Reply:
no yea, I'm completely ok with having a limit, my point is this limit being different for single file commits and multiple files commit. could be the same limit for both :)